### ISSUE TITLE
[ui] Less obtrusive diagnostic outputs

### ISSUE THREAD

**Issue by paullewis** [google/labs-prototypes#241]
Diagnostic / debug outputs from – say – LLMs are displayed prior to the output node. It's likely that we can make this a part of the output node itself rather than a separate node.

Might need some wiring changes to embed the diagnostic info into the output node itself, maybe? I looked at the output info coming through and it does seem like it's only the first candidate. 

/cc @dglazkov

**Comment by dglazkov** [comment]
Related, not sure if worth it: I had a somewhat crazy idea of retaining a flow diagram as the board runs. Right now, it just shows various nodes it encounters as changing text, but that text is gone never to be seen again.

What if instead, it was collecting them into a tiny little horizontal sequence that is minimized by default, but could expanded to see what nodes were travelled as the graph ran?

And maybe even click on them to see various diagnostic bits?

**Comment by dglazkov** [comment]
> make this a part of the output

At least in my mind, those seem somewhat orthogonal to the output -- they are intermediate results that are mostly useful for diagnostics.

**Comment by dglazkov** [comment]
Some running notes: I am noticing that I am downloading the diagram a lot. It's too small to see the actual wiring, and I end up having lots of pngs of diagrams.

**Comment by paullewis** [comment]
I think the short term fix I'll make today is to make the diagram 100% width and therefore larger. The longer term fix, at least in my head, is probably to make the graph the star of the show. I intuitively reach for it to understand what's going on and where, more so than the log we currently emit (though it has its uses).

**Comment by dglazkov** [comment]
running notes:

- omg, seeing nodes run on board is a massive step up in debuggability. I already adapted to add one node at a time and seeing whether I wired it right or not on refresh
- the failures that seem to get me every time are all related to missing wires: misnaming outputs/inputs, miswiring to the wrong node, and creating cycles where I didn't mean to
- I use DevTools console a lot to spelunk through inputs/outputs and review the sequence of runs

**Comment by paullewis** [comment]
FWIW I am planning on adding:

1. A "node inspector" of sorts so that you can see the inputs & outputs listed for each node.
2. A history so that you can see the run sequence in the UI. Ultimately I'd like this to have save/restore functionality, too.
3. An artificial delay so that you can see the sequence run
4. A 'reset board' button so that you can return to a fresh state

**Comment by dglazkov** [comment]
Now that I see the nodes light up, I really want to teach boards to walk into invoked boards:
#197 Otherwise, for graphs like https://breadboard-ai.web.app/?board=%2Fgraphs%2Fboard-caller.json, most of the node-walking is hidden from the harness.

**Comment by dglazkov** [comment]
> presume it's mostly a case of bubbling up everything from the board's harness

yup. The machinery described in #197 is one of the possible solutions. A short-term hack might be to build a nicer `Probe`. The way they work today, do get passed into the nested graphs.

> if they are namespaced in some way

Each graph is a clean namespace slate, so it will be super-tricky to make them light up on the diagram.

**Comment by dglazkov** [comment]
Running notes:

- I find myself spending a lot of time spelunking through the console to find what inputs/outputs of a node after a run.
